lib/hanami/helpers/form_helper/form_builder.rb

```diff
@@ -766,6 +766,45 @@ def select(name, values, attributes = {})
           end
         end
 
+        # Datalist input
+        #
+        # @param name [Symbol] the input name
+        # @param values [Hash] a Hash to generate <tt><option></tt> tags of datalist.
+        #   Keys correspond to <tt>value</tt> and values correspond to the content.
+        # @param list [String] the name of list for the text input, it's also the id of datalist
+        # @param attributes [Hash] HTML attributes to pass to the input tag
+        #
+        # @since 0.3.1
+        #
+        # @example Basic usage
+        #   <%=
+        #     # ...
+        #     values = Hash['it' => 'Italy', 'us' => 'United States']
+        #     datalist :stores, values, 'books'
+        #   %>
+        #
+        #   # Output:
+        #   #  <input type="text" name="book[store]" id="book-store" list="books">
```

Comment:
I think we should discuss about the generation if `<input>` tag. To me it makes the helper complex, options like `options`, `datalist` to manipulate the output are clunky compare to explicit and verbose declaration of nested `input` then `datalist` tags